### ISSUE TITLE
Prevent sign-ups from under-13 users

### ISSUE THREAD

**Issue by eliascodes** [TwinePlatform/twine-visitor#584]
Related to TwinePlatform/hq#40

1. Prevent sign-ups of users whose year of birth (if entered) makes them under 13.
1. On sign-ups that don't enter year of birth, require a checkmark declaration that they are old enough.

**Comment by astroash** [comment]
@eliascodes year of birth is currently not optional. Is this issue implying that it should be? 

If so the following additional changes will need to be made:
- allow optional birthYear field on server
- add data aggregates for 'prefer not to say' birthYear

**Comment by eliascodes** [comment]
Can be either:
1. Optional field, defaulting to 'prefer not to say'
1. Non-optional field, with an option of 'prefer not to say'

On the data-aggregates, it does kinda fuck with how we're currently specifying age ranges. The unsatisfying but simplest option is I guess sending 'prefer not to say' as the value for 'age' in the filter query.

What do you think?

**Comment by astroash** [comment]
Yeah, I agree. Just wanted to double check knock on effect of implementing this. It's why I abandoned it last time I picked this up.
